pkg/controllers/routercerts/controller.go

```diff
@@ -103,21 +126,27 @@ func (c *routerCertsDomainValidationController) validateRouterCertificates() ope
 	}
 
 	// get router certs secret
-	secret, err := c.secretLister.Secrets(c.targetNamespace).Get(c.secretName)
+	var secret *corev1.Secret
+	secret, c.usingDefaultSecret, err = common.GetActiveRouterSecret(c.secretLister, c.defaultSecretNamespace, c.defaultSecretName, c.customSecretNamespace, c.customSecretName)
 	if err != nil {
-		return newRouterCertsDegradedf("NoRouterCertSecret", "secret/%v -n %v: could not be retrieved: %v", c.secretName, c.targetNamespace, err)
+		return newRouterCertsDegradedf("NoRouterCertSecret", "secret/%v -n %v: could not be retrieved: %v", c.getActiveSecretName(), c.getActiveSecretNamespace(), err)
+	}
+
+	certKey := ingress.Spec.Domain
+	if !c.usingDefaultSecret {
+		certKey = corev1.TLSCertKey
```

Comment:
looks more like you need a unified way to retrieve the secret's cert and key

Reply:
Added a method to the common package